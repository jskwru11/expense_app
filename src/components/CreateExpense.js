import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expensesAction';


export class CreateExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(CreateExpense)

// const CreateExpense = (props) => (
//     <div>
//         <h1>Add Expense</h1>
//         <ExpenseForm 
//             onSubmit={(expense) => {            
//                 props.dispatch(startAddExpense({...expense}));
//                 props.history.push('/');
//             }}
//         />
//     </div>
// );




// export default connect()(CreateExpense);