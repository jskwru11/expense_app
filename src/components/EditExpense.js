import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expensesAction';



const EditExpense = (props) => {
    return (
        <div>
            <ExpenseForm
                expense={props.expenses} 
                onSubmit={(expense) => {
                    // Dispatch the action to edit the expense
                    props.dispatch(editExpense(props.match.params.id,{...expense}));
                    // Redirect to the dashboard
                    props.history.push('/');
                }}
            />
            <button onClick={(e) => {
                props.dispatch(removeExpense({id: props.match.params.id}))
                props.history.push('/');
            }}>Remove</button>
        </div>
    );
}

const mapStateToProps = (state, props) => {
    return {
        expenses: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};


export default connect(mapStateToProps)(EditExpense);