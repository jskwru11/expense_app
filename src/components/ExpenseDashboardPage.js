import React from 'react';
import ExpenseList from './expenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpenseSummary';


const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
   
);

export default ExpenseDashboardPage;