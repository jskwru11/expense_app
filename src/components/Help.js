import React from 'react';

const Help = () => (
    <div>This is from my help page</div>
);

export default Help;