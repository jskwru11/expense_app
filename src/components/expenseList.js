import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/selectors';



const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map((expense) => (
            <ExpenseListItem key={expense.id} {...expense}/>
        ))}
        
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    };
};

//create HOC for connected components
export default connect(mapStateToProps)(ExpenseList);
