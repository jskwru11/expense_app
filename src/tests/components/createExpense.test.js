import React from 'react';
import { shallow } from 'enzyme';
import { CreateExpense } from '../../components/CreateExpense';
import expenses from '../fixtures/expenses';

let startAddExpense, history, wrapper;

beforeEach(() => {
    startAddExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<CreateExpense startAddExpense={startAddExpense} history={history} />);
});

test('should render CreateExpense correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSumbit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(startAddExpensee).toHaveBeenLastCalledWith(expenses[1]);
});
