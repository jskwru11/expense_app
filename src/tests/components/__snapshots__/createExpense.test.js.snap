// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreateExpense correctly 1`] = `undefined`;
